test/test_core.py

```diff
@@ -205,6 +205,49 @@ def test_core_uninstall(run_command):
     assert not _in(result.stdout, "arduino:avr")
 
 
+def test_core_uninstall_tool_dependency_removal(run_command):
+    # These platforms both have a dependency on the arduino:avr-gcc@7.3.0-atmel3.6.1-arduino5 tool
+    # arduino:avr@1.8.2 has a dependency on arduino:avrdude@6.3.0-arduino17
+    assert run_command("core install arduino:avr@1.8.2")
+    # arduino:megaavr@1.8.4 has a dependency on arduino:avrdude@6.3.0-arduino16
+    assert run_command("core install arduino:megaavr@1.8.4")
+    assert run_command("core uninstall arduino:avr")
+
+    result = run_command("config dump --format json")
+    assert result.ok
+    settings_json = json.loads(result.stdout)
+    arduino_tools_path = os.path.join(
+        settings_json["directories"]["data"],
+        "packages",
+        "arduino",
+        "tools"
+    )
```

Comment:
There's no need to load the settings like so, you already have access to these paths through the `data_dir`, `working_dir` and `downloads_dir` fixtures, you can find them on the `tests/conftest.py` file, they're also used for some tests in this file.

I also suggest `pathlib.Path` to join paths and check those files exist.

Reply:
Thanks for your advice Silvano. I see you already noticed I updated the PR accordingly, so I'll mark this "Resolved".